src/main/java/uk/nhs/digital/nhsconnect/nhais/mapper/AcceptanceCodeMapper.java

```diff
@@ -0,0 +1,25 @@
+package uk.nhs.digital.nhsconnect.nhais.mapper;
+
+import org.hl7.fhir.r4.model.Parameters;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.AcceptanceCode;
+
+public class AcceptanceCodeMapper implements FromFhirToEdifactMapper<AcceptanceCode> {
+    private final static String ACCEPTANCE_CODE = "acceptanceCode";
+
+    public AcceptanceCode map(Parameters parameters) {
+        return AcceptanceCode.builder()
+            .code(getAcceptanceCode(parameters))
+            .build();
+    }
+
+    private String getAcceptanceCode(Parameters parameters) {
+        return parameters.getParameter()
+            .stream()
+            .filter(param -> ACCEPTANCE_CODE.equals(param.getName()))
+            .map(Parameters.ParametersParameterComponent::getValue)
+            .map(Object::toString)
+            .filter(AcceptanceCode::isCodeAllowed)
+            .findFirst()
+            .orElseThrow(() -> new IllegalStateException("Acceptance Code not supported"));
```

Comment:
or just do `orElseThrow()` Using streams we are not able to break in the middle and print what's wrong (easily and without splitting it). Simple throw is enough I think.

Reply:
Please throw a FhirValidationException and identify which parameter is missing (by name) so that a correct OperationOutcome response is produced by the API